### ISSUE TITLE
New `ZipPageLoader` breaks "Split wide pages"

### ISSUE THREAD

**Issue by Shamicen** [mihonapp/mihon#479]
### Steps to reproduce

Try opening a `CBZ` or `Zip` file, containing a wide image, with `Split wide pages` enabled (should also be broken for rar and epub but untested).

### Expected behavior

Should load.

### Actual behavior

Doesn't load.

### Crash logs

```
java.io.IOException: Resetting to invalid mark
at java.io.BufferedInputStream.reset(BufferedInputStream.java:450)
at tachiyomi.core.common.util.system.ImageUtil.extractImageOptions(ImageUtil.kt:559)
at tachiyomi.core.common.util.system.ImageUtil.extractImageOptions$default(ImageUtil.kt:550)
at tachiyomi.core.common.util.system.ImageUtil.isWideImage(ImageUtil.kt:128)
at eu.kanade.tachiyomi.ui.reader.viewer.pager.PagerPageHolder.process(PagerPageHolder.kt:198)
at eu.kanade.tachiyomi.ui.reader.viewer.pager.PagerPageHolder.access$process(PagerPageHolder.kt:33)
at eu.kanade.tachiyomi.ui.reader.viewer.pager.PagerPageHolder$setImage$2.invokeSuspend(PagerPageHolder.kt:144)
at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
```

### Mihon version

newest debug version

### Android version

Android 12

### Device

Pixel 5 emulator

### Other details

The cause of this bug is, that ZipFiles created from channels do not support `mark` and `reset` (their streams do return true when using `markSupported()` but they actually don't). 

Split wide images uses ImageUtils `extractImageOptions()`, which needs the stream to support reset to work.

Possible solutions: 

1. pass the stream function to `extractImageOptions()`. (I tested this and it works. Performance did seem to be terrible, but that could just be my emulator.)
2. read the stream into a ByteArray in `setImage()` and use a buffered `byteArrayInputStream` from there. (I tested this as well and performance seemed to be OK but I can't really say anything definitve, because my emulator performance is not the best.) 
3. Pass the ZipFile and its entry to `extractImageOptions()`. We would probably need to do that for `Rar` and `Epub` files as well so this is probably not a good solution. (this should in theory have the exact same effect as 1 but the performance was a lot better in the limited tests I ran, but like i said it could just have been my emulator being weird.)

I could write a fix myself, but I think it would be better if someone that can test the performance impact would do it instead / give feedback about the best performing solution.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have gone through the [FAQ](https://mihon.app/docs/faq/general) and [troubleshooting guide](https://mihon.app/docs/guides/troubleshooting/).
- [X] I have updated the app to version **[0.16.4](https://github.com/mihonapp/mihon/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I will fill out all of the requested information in this form.

**Comment by FooIbar** [comment]
This is because the [`available()`](https://developer.android.com/reference/java/util/zip/InflaterInputStream#available()) method of `InflaterInputStream` always returns 1 if EOF has not been reached.
So it can't be used for mark limit.
https://github.com/mihonapp/mihon/blob/d8b9a9f593911569ff2bceb49b4f020978d0d2e1/core/common/src/main/kotlin/tachiyomi/core/common/util/system/ImageUtil.kt#L554
Change to `Int.MAX_VALUE` here would fix it.

**Comment by Shamicen** [comment]
Closed in favor of #485
